edit/MPE-243: fix create company doc (#285)
mia-platform/documentation: docs/development_suite/company/create.mdx

Within a company, it is possible to:
- configure the infrastructure on which new projects will be based; 
- analyze the configuration of all existing projects;
- manage users and permissions.

#### Saas
To create a new company in the Mia-Platform SaaS Console, open a [Service Request](https://makeitapp.atlassian.net/servicedesk/customer/portal/21/group/79/create/340) and the company will be created for you.

#### On-Premise 
Before creating a new company in a Mia-platform On-Premise Console installation, contact your Mia-Platform referent.
Once you made sure to have all the necessary requirements for creating a new company, you can continue with this guide.

:::caution
Remember that the possibility of adding new companies may depend on the type of license you have subscribed to. 

Only a limited amount of companies may be available for creation within your license terms.
:::

<br/>

The next steps will guide you through all the processes to create and correctly configure a new company.
 
### Requirements

To create a company, you must have:

- access to the Console Backoffice (granted by the `console_cms` group)

> - `console.root.company.create` permission

Author: Sure, I can add this information. @fredmaggiowski , do you know better about Davide's question?

## Create a company from API

> You can create a new company by making a POST request to the `/api/backend/companies` API. 

Reviewer: l'host della console cambia tra on premise e paas. Vogliamo specificare che ci vuole l'host? 

Mettiamo un esempio? di curl? 

```suggestion
You can create a new company by making a POST request to the `${THE_CONSOLE_HOST}/api/backend/companies` API. 
```

:::info
This API is visible inside the Console API Portal, under the `Companies` tag.
:::

You will be asked to set the **name** and the **description** of the company to add (only name is required). 

In addition, you will be added as the first Company user with the role of [`Company Owner`](/docs/development_suite/identity-and-access-management/console-levels-and-permission-management#users-capabilities-inside-console).

> ## Create a company from CMS

Author: Can try that!

It is also possible to manually create a company from the Console Backoffice. 

However, you will lose some of the automations provided by the company creation API, such as adding the first user, or providing a company unique identifier.

:::danger
It is recommended to use the API to create a new company. Use this method only if strictly necessary.

> :::

Author: Do you think placing it between the API paragraph and this one could work for you?

> :::

Reviewer 2: It would probably be more effective to place it directly under the "Create a company from CMS" title 😁

    "type": "azure-devops",
    "visibility": "private",
    "basePath": "mia-platform-devops"
  }
  ```

- **pipelines** (_required_):
  
  An object that specifies the information regarding the CI/CD pipelines used by the company. 

  :::info
  This field can also be set by [selecting a default CI/CD tool provider](/docs/development_suite/set-up-infrastructure/providers-management#set-a-default-provider) in the **Providers** section of the Console.
  :::
  
  The object must have the following properties:

  - **providerId** (_required_): unique identifier of the CI/CD tool provider associated with the company.
    
    It corresponds to the `providerId` field of the provider's CRUD collection and is used to retrieve information about the used provider. 

  - **type** (_required_): one of the supported provider types. Check out the complete list of [supported providers](/docs/development_suite/set-up-infrastructure/providers-management);

  :::note
  The following properties are supported only when the provider type is **"azure-devops"**:

    - **azurePipelineId**: TODO
  
  :::

  :::note
  The following properties are supported only when the provider type is **"jenkins"**:

>     - **options**: TODO

Reviewer 3: This is an object with view and name:

```json
{
  "view": {
    "name": ""
  }
}
```

and will be used during the generation of the service to name the Jenkins pipeline on the correct view

  
  :::

  ##### Example Configuration
  ```json
  {
    "providerId": "my-gitlab",
    "type": "gitlab-ci"
  }
  ```
    
- **environmentsVariables**:
   
  An object that describes the configuration to enable the Variables section in the Project Settings area of the Console.

  :::info
  This field can also be set by [selecting a default secret manager provider](/docs/development_suite/set-up-infrastructure/providers-management#set-a-default-provider) in the **Providers** section of the Console.
  :::

  :::caution
  This field is not required. However, by leaving an empty configuration you will need to manually configure a secret manager provider for each project you create.
  :::

  The object can be configured with the following properties:

>   - **providerId** (_required_): unique identifier of the secret manager provider associated with the company. 

Reviewer 3: This is not required, for example using gitlab which is the same git provider

  - **type** (_required_): one of the supported provider types. Check out the complete list of [supported providers](/docs/development_suite/set-up-infrastructure/providers-management);

  :::note
  The following properties are supported only when the provider type is **"gitlab"**:

  - **baseUrl**: base path where the variables will be read from. 

    If not specified, variables will be read from the first parent group of the Configurations project in GitLab.

    As an example, if your GitLab project is saved in **`https://git.tools.mia-platform.eu/mia-platform/Test/configurations`**, the environment variables will be written in the **`mia-platform/Test`** group.

  - **storage**: additional settings related to the storage of environment variables within the GitLab installation.

    It is composed by the following properties:

    - **id**: TODO

    - **type**: TODO

    - **path**: Either one of the following:

      - `groups`: TODO
      - `projects`: TODO

    <br/>
    
    :::info
    This configuration is saved only within the company model, and is retrieved by projects at runtime. 

>     If in your company all the environment variables are set to be located in the parent group, by editing this field you can safely change the variables path for all the projects in the company in one go.

Reviewer 3: ```suggestion
    If in your company all the environment variables are set to be located in the parent group, by editing this field you can safely change the variables path for all the projects in the company.
```

    :::

  :::

  ##### Example Configuration
  ```json
  {
    "providerId": "my-gitlab",
    "type": "gitlab",
    "baseUrl": "https://my-gitlab-host",
    "storage": {
      "type": "projects",
      "path": "clients/mia-platform/configurations"
    }
  }
  ```

- **environments** (_required_):
  
  An array of objects containing the definitions of the environments for the company.

  :::note

>   The content of this array will be interpolated during project creation by replacing `%projectId%` strings with the `projectId` field specified in the project.

Reviewer 3: Can we check if it's only projectId? I don't think so

  :::

  Each object is shaped as follows:

   - **label**: descriptive and human-readable identifier associated with the environment;  
   - **envId**: unique identifier associated with the environment;
   - **envPrefix**: defines a common prefix associated with environment-specific variables;
   - **description**: short description of the environment;
   - **isProduction**: boolean value indicating whether it is considered to be a production environment;

>    - **hosts**: TODO

Reviewer 3: An array of object, which specify at the project creation which host will be exposed (if exposed in template), used by links in environment sections

Something like:

```json
    "hosts": [
      {
        "host": "test.%projectId%.console.gcp.mia-platform.eu",
        "isBackoffice": false,
        "scheme": "https"
      }
    ],
```

Where isBackoffice and scheme are optional.

>    - **links**:     TODO

Reviewer 3: This is not set inside company if I well remember

>    - **cluster**:     TODO

Reviewer 3: Something like:

```json
    "cluster": {
      "clusterId": "objectId-of-the-cluster",
      "namespace": "%projectId%-dev",
      "kubeContextVariables": {
        "KUBE_URL": "",
        "KUBE_TOKEN": "",
        "KUBE_CA_PEM": ""
      }
    }
```

>    - **dashboard**:     TODO

Reviewer 3: This is not from the company side, but are taken from the project template

   - **deploy**: An object that describes the configuration of CI/CD pipelines for the environment.
    
    :::info
    This field overrides the configuration provided in the `pipelines` field for a specific environment. 
    :::

    It can be configured following the configuration of the `pipelines` company field.

>     TODO jobId, statusWebhookSecretCredentialsId, paramsMap

Reviewer 3: jobId and paramsMap are set for Jenkins

statusWebhookSecretCredentialsId is set by github-actions, but during the project creation it should not be here. But it should be verified, I'm not sure

  ```

  :::danger
  The values saved here are not encrypted!

  Do not set the `cluster.kubeContextVariables` object with plain values to access the cluster.

  Instead, prefer the use of an interpolated value. By specifying a variable key name (as shown in the example), the value will be taken from the corresponing project variable. 
  :::

- **logicalScopeLayers**:
   
  An array of objects identifying the set of logical layers available in the current company. 
  
  Each object is shaped as follows:

  - **name**: identifies the name of the logical layer;
  - **order**: identifies the numerical order of the layer, used to display layers according to the sorting order defined by the user.

  The `logicalScopeLayers` field will be created in the project model according to the project `layerId` value. Please refer to the [Create a Project](docs/development_suite/set-up-infrastructure/create-project) section for further details.

  ##### Example Configuration
  ```json
  [
    { "name": "layer-1", "order": 2}, 
    { "name": "layer-2", "order": 1}
  ]
  ```

- **availableNamespaces**:
  
  Namespaces internally accessible from your project, using the `cross-projects` endpoint. 
  

>   Namesapces are useful when your company is made up of several projects that communicate with each other.

Reviewer 3: ```suggestion
  Namespaces are useful when your company is made up of several projects that communicate with each other.
```

Only a limited amount of companies may be available for creation within your license terms.
:::

Once you made sure you meet all the necessary requirements to create a new company, you can continue with this guide.

<br/>

The next steps will guide you through all the processes to create and correctly configure a new company.
 
### Requirements

To create and configure a company, you must have:

- access to the Console Backoffice (granted by the `console_cms` group)
- `console.root.company.create` permission

## Create a company from API

You can create a new company by making a POST request to the `/api/backend/companies` API. 

:::info
This API is visible inside the Console API Portal, under the `Companies` tag.
:::

You will be asked to set the **name** and the **description** of the company to add (only name is required). 

In addition, you will be added as the first Company user with the role of [`Company Owner`](/docs/development_suite/identity-and-access-management/console-levels-and-permission-management#users-capabilities-inside-console).

## Create a company from CMS

> It is also possible to manually create a company from the Console Backoffice. 

Reviewer 4: Nel titoplo si parla di CMS e qui di Backoffice, io utilizzerei lo stesso acronimo. 
Fiontanto che il Backoffice non sostituisce a pieno il CMS mi riferirei sempre al CMS

Author: ok :+1:

Regardless, head over to the CMS page and click the `Bindings` menu item.

:::caution
Only a restricted set of users have access to the users management section in CMS, ask your Console Administrator if you need the right permissions to access this functionality.
:::

In the `Bindings` section, click the `Add new` button and fill in the form:
- type a custom **Binding ID**

  :::danger
  The _Binding ID_ must be unique in the whole collection!  

  As a general recommendation we suggest `{companyId}-{desiredRole}`, however, in some cases a more complex structure may be necessary.
  :::

- select the proper **Role** (since it is the first user, it is recommended to set the role [`Company Owner`](/docs/development_suite/identity-and-access-management/console-levels-and-permission-management#users-capabilities-inside-console).)

- select the correct user as **Subject**

- assign the correct **Resource** by filling it with an object as follows:

  ```json
  {
    "resourceType": "company",
    "resourceId": "{THE COMPANY ID YOU PROVIDED UPON CREATION}"
  }
  ```
## Company Configuration

Once you create a company, you can configure:

> - providers (git providers, secret managers, CI/CD tools)

Reviewer 4: Dato che questo probabilmente è il primo accenno ai provider in un flusso utente spiegherei brevissimamente cosa intendiamo noi come providers e per cosa vengono utilizzati nella console. Magari un primo utente non ne hai idea e non collega leggendo un generico "provider"

> - clusters

Reviewer 4: Metterei come ordine per primi i cluster rispetto ai provider, giusto per ordine logico

Author: ok :+1:

Reviewer 2: Nel caso invertirei anche sotto i relativi paragrafi

      - /groupA/groupB -> if A does not exist, then it will NOT be created any group.

    </TabItem>

    <TabItem value="github" label="GitHub">

      It corresponds to the **path of an existing organization**. 
      
      If the path does not exist, the project will not be created.

      For example, if your organization is reachable at the URL **`https://github.com/mia-platform`**, then the `basePath` must be **`mia-platform`**

    </TabItem>

    <TabItem value="azure-devops" label="Azure DevOps">

      It corresponds to the **path where your Azure project is reachable**: the pattern is _company-name/project-name_.
      
      For example, if you project is reachable at the URL **`https://dev.azure.com/mia-platform-devops/Test`**, then the `basePath` must be **`mia-platform-devops/Test`**.

    </TabItem>

    <TabItem value="bitbucket" label="BitBucket">

      It corresponds to the **path where your BitBucket project is reachable**: the pattern is _projects/project-code_.

      For example, if you project is reachable at the URL **`https://bitbucket.mia-platform.eu/projects/Test`**, then the `basePath` must be **`projects/Test`**.

    </TabItem>

    </Tabs>

>   ##### Example Configuration

Reviewer 4: Qui, giusto per essere a prova di super-dummy, farei un esempio per ogni provider mettendo tipo le tab che abbiamo già da altra parti.

    "basePath": "mia-platform-devops"
  }
  ```

- **pipelines** (_required_):
  
  An object that specifies the information regarding the CI/CD pipelines used by the company. 

  :::info
  This field can also be set by [selecting a default CI/CD tool provider](/docs/development_suite/set-up-infrastructure/providers-management#set-a-default-provider) in the **Providers** section of the Console.
  :::
  
  The object must have the following properties:

  - **providerId** (_required_): unique identifier of the CI/CD tool provider associated with the company.
    
    It corresponds to the `providerId` field of the provider's CRUD collection, and it is used to retrieve information about the provider. 

  - **type** (_required_): one of the supported provider types. Check out the complete list of [supported providers](/docs/development_suite/set-up-infrastructure/providers-management);

  :::note
  The following properties are supported only when the provider type is **"azure-devops"**:

    - **azurePipelineId**: TODO
  
  :::

  :::note
  The following properties are supported only when the provider type is **"jenkins"**:

    - **options**: TODO
  
  :::

>   ##### Example Configuration

Reviewer 4: Anche qui idem, mettere un esempio per ogni provider supportato

- type a custom **Binding ID**

  :::danger
  The _Binding ID_ must be unique in the whole collection!  

  As a general recommendation we suggest `{companyId}-{desiredRole}`, however, in some cases a more complex structure may be necessary.
  :::

- select the proper **Role** (since it is the first user, it is recommended to set the role [`Company Owner`](/docs/development_suite/identity-and-access-management/console-levels-and-permission-management#users-capabilities-inside-console).)

- select the correct user as **Subject**

- assign the correct **Resource** by filling it with an object as follows:

  ```json
  {
    "resourceType": "company",
    "resourceId": "{THE COMPANY ID YOU PROVIDED UPON CREATION}"
  }
  ```
## Company Configuration

Once you create a company, you can configure:

- providers (git providers, secret managers, CI/CD tools)
- clusters
- projects default configurations (such as environments, logical layers and accessible namespaces)

:::info
Each of these configurations will be applied as the default configuration to newly created projects within the company.
:::

Let's now proceed with each step of the configuration:

### Configure Providers

> Git providers, Secret Managers and CI/CD tools can be configured from the **Providers** section inside Company Overview area of the Console.

Reviewer 2: ```suggestion
Git Providers, Secret Managers, and CI/CD Tools can be configured from the **Providers** section within the Company Overview area of the Console.
```

For further details on how to create new providers, you can follow the [providers configuration guide](/docs/development_suite/set-up-infrastructure/configure-provider.mdx).

> ### Connect a cluster

Reviewer 2: ```suggestion
### Connect a Cluster
```

> Clusters can be connected to a company from the **Clusters** section inside Company Overview area of the Console.

Reviewer 2: ```suggestion
Clusters can be connected to a company from the **Clusters** section within the Company Overview area of the Console.
```

:::info
This configuration can be done from the `Companies` section of the Console Backoffice.
:::

The projects default configurations fields are:

- **defaultTemplateId**:
  
  The identifier of the default template to be used for the creation of a project. 
  This information can be changed for a specific project during the steps of the project creation wizard.

- **cmsImageName**:
  
  The name of the Docker image used to generate the CMS of a project. It should also contain the CMS tag to be used.
  This value will interpolate the template archive of your project.
  
  if `cms-site` service is disabled during the project creation, this field will not be used.

- **repository** (_required_):
  
  An object that specifies the information regarding the repository where the projects of this company will be created. 
  
  :::info
  Both _providerId_ and _type_ fields can also be set by [selecting a default git provider](/docs/development_suite/set-up-infrastructure/providers-management#set-a-default-provider) in the **Providers** section of the Console.
  :::
  
  The object must have the following properties:

>   - **providerId** (_required_): unique identifier of the git provider associated with the company 

Reviewer 2: ```suggestion
  - **providerId** (_required_): unique identifier of the Git Provider associated with the company.
```

  
    It corresponds to the `providerId` field of the provider's CRUD collection, and it is used to retrieve information about the provider. 

  - **type** (_required_): one of the supported provider types. Check out the complete list of [supported providers](/docs/development_suite/set-up-infrastructure/providers-management);
  
  - **visibility**: visibility of the project. Either one of the following:

    - `public`: the repository can be accessed and viewed by anyone, including non-authenticated users;

>     - `internal`: the repository can be accessed and viewed by any authenticated user within the same git provider instance or organization;

Reviewer 2: ```suggestion
    - `internal`: the repository can be accessed and viewed by any authenticated user within the same Git Provider instance or organization;
```

    - `private`: the repository can only be accessed and viewed by users who have been explicitly granted access by the repository owner.
  
  - **basePath**: base path where the project will be created. 
    
    :::caution
    Make sure to have the right permissions for creating projects on the configured path.
    :::

>     Here some tips on how to correctly configure the _basePath_ property, grouped by Git provider:

Reviewer 2: ```suggestion
    Here some tips on how to correctly configure the _basePath_ property, grouped by Git Provider:
```

    <Tabs>

    <TabItem value="gitlab" label="GitLab" default>

      It corresponds to the **basePath of the group** where to create the project.

      If the group does not exist, the project will be created only if the user has the needed permission to create the group in its parent.

      For example, if your group is reachable at the URL **`https://git.tools.mia-platform.eu/mia-platform/projects`**, 
      then the `basePath` must be **`mia-platform/projects`**.
      
      If your group (B) does not exist:

      - /groupA/groupB -> if user have permission to create groups in A, then B will be created;
      - /groupA/groupB -> if user have no permission to create groups in A, then B will NOT be created;
      - /groupA/groupB -> if A does not exist, then it will NOT be created any group.

    </TabItem>

    <TabItem value="github" label="GitHub">

      It corresponds to the **path of an existing organization**. 
      
      If the path does not exist, the project will not be created.

>       For example, if your organization is reachable at the URL **`https://github.com/mia-platform`**, then the `basePath` must be **`mia-platform`**

Reviewer 2: ```suggestion
      For example, if your organization is reachable at the URL **`https://github.com/mia-platform`**, then the `basePath` must be **`mia-platform`**.
```

    </TabItem>

    <TabItem value="azure-devops" label="Azure DevOps">

      It corresponds to the **path where your Azure project is reachable**: the pattern is _company-name/project-name_.
      
      For example, if you project is reachable at the URL **`https://dev.azure.com/mia-platform-devops/Test`**, then the `basePath` must be **`mia-platform-devops/Test`**.

    </TabItem>

>     <TabItem value="bitbucket" label="BitBucket">

Reviewer 2: ```suggestion
    <TabItem value="bitbucket" label="Bitbucket">
```

>       It corresponds to the **path where your BitBucket project is reachable**: the pattern is _projects/project-code_.

Reviewer 2: ```suggestion
      It corresponds to the **path where your Bitbucket project is reachable**: the pattern is _projects/project-code_.
```

      For example, if you project is reachable at the URL **`https://bitbucket.mia-platform.eu/projects/Test`**, then the `basePath` must be **`projects/Test`**.

    </TabItem>

    </Tabs>

  ##### Example Configuration
  ```json
  {
    "providerId": "my-azure-devops",
    "type": "azure-devops",
    "visibility": "private",
    "basePath": "mia-platform-devops"
  }
  ```

- **pipelines** (_required_):
  
  An object that specifies the information regarding the CI/CD pipelines used by the company. 

  :::info
  This field can also be set by [selecting a default CI/CD tool provider](/docs/development_suite/set-up-infrastructure/providers-management#set-a-default-provider) in the **Providers** section of the Console.
  :::
  
  The object must have the following properties:

>   - **providerId** (_required_): unique identifier of the CI/CD tool provider associated with the company.

Reviewer 2: ```suggestion
  - **providerId** (_required_): unique identifier of the CI/CD Tool associated with the company.
```

    
    It corresponds to the `providerId` field of the provider's CRUD collection, and it is used to retrieve information about the provider. 

  - **type** (_required_): one of the supported provider types. Check out the complete list of [supported providers](/docs/development_suite/set-up-infrastructure/providers-management);

  :::note
  The following properties are supported only when the provider type is **"azure-devops"**:

    - **azurePipelineId**: TODO
  
  :::

  :::note
  The following properties are supported only when the provider type is **"jenkins"**:

    - **options**: TODO
  
  :::

  ##### Example Configuration
  ```json
  {
    "providerId": "my-gitlab",
    "type": "gitlab-ci"
  }
  ```
    
- **environmentsVariables**:
   
  An object that describes the configuration to enable the Variables section in the Project Settings area of the Console.

  :::info

>   This field can also be set by [selecting a default secret manager provider](/docs/development_suite/set-up-infrastructure/providers-management#set-a-default-provider) in the **Providers** section of the Console.

Reviewer 2: ```suggestion
  This field can also be set by [selecting a default Secret Manager](/docs/development_suite/set-up-infrastructure/providers-management#set-a-default-provider) in the **Providers** section of the Console.
```

  :::

  :::caution

>   This field is not required. However, by leaving an empty configuration you will need to manually configure a secret manager provider for each project you create.

Reviewer 2: ```suggestion
  This field is not required. However, by leaving an empty configuration you will need to manually configure a Secret Manager for each project you create.
```

  :::

  The object can be configured with the following properties:

>   - **providerId** (_required_): unique identifier of the secret manager provider associated with the company. 

Reviewer 2: ```suggestion
  - **providerId** (_required_): unique identifier of the Secret Manager associated with the company.
```

  
    It corresponds to the `providerId` field of the provider's CRUD collection, and it is used to retrieve information about the provider. 

  - **type** (_required_): one of the supported provider types. Check out the complete list of [supported providers](/docs/development_suite/set-up-infrastructure/providers-management);

  :::note
  The following properties are supported only when the provider type is **"gitlab"**:

  - **baseUrl**: base path where the variables will be read from. 

    If not specified, variables will be read from the first parent group of the Configurations project in GitLab.

    As an example, if your GitLab project is saved in **`https://git.tools.mia-platform.eu/mia-platform/Test/configurations`**, the environment variables will be written in the **`mia-platform/Test`** group.

  - **storage**: additional settings related to the storage of environment variables within the GitLab installation.

    It is composed by the following properties:

    - **id**: TODO

    - **type**: TODO

>     - **path**: Either one of the following:

Reviewer 2: ```suggestion
    - **path**: either one of the following:
```

  - **description**: short description of the environment;
  - **isProduction**: boolean value indicating whether it is considered to be a production environment;
  - **hosts**: an array of objects identifying the hosts used by the environment.
  
    Each host is shaped as follows:

    - **host**: URL identifying where the host is located;
    - **isBackoffice**: boolean value indicating whether the host is a backoffice instance;
    - **scheme**: the URL schema, typically either _http_ of _https_.

    This object can be used to specify where a specific environment should be hosted,
    where it should expose its documentation or where to access its CMS.

  - **links**: an array of objects containing a series of useful links.

    Each link is shaped as follows:

    - **id**: unique identifier associated with the link;
    - **url** (_required_): link redirection URL.

  - **dashboards**: an array of objects identifying environments metrics and performance indicators.

    Each dashboard is shaped as follows:

      - **id** (_required_): unique identifier associated with the dashboard;
      - **label** (_required_): descriptive human-readable identifier for the dashboard;  

>       - **url** (_required_): URL where the metrics are retreived from;

Reviewer 2: ```suggestion
      - **url** (_required_): URL from which the metrics are retrieved;
```

      - **type** (_required_): defines the integration method of the dashboard in the Console.
  
        Either one of the following: 
        
        - **iframe**: the dashboard will be included within the Console UI;
        - **newTab**: the dashboard will open on a new browser tab.

>       - **category**: indicates the group in which the dashboard will be presented in the console UI.

Reviewer 2: ```suggestion
      - **category**: indicates the group in which the dashboard will be presented in the Console UI.
```

  
  - **environmentsVariables**: An object that describes the configuration of variables for the environment.
    
    :::info
    This field overrides the configuration provided in the `environmentsVariables` field for a specific environment. 
    :::

    It can be configured following the configuration of the `environmentsVariables` company field.

>   - **deploy**: An object that describes the configuration of CI/CD pipelines for the environment.

Reviewer 2: ```suggestion
  - **deploy**: an object that describes the configuration of CI/CD pipelines for the environment.
```

      - **hostname**:        TODO
      - **port**:        TODO
      - **namespace**:        TODO
      - **clusterId**:        TODO
      - **accessToken**:        TODO
      - **base64CA**:        TODO
      - **proxy**:        TODO
      - **serviceAccount**:        TODO

  <br/>

  ##### Example Configuration
  ```json
    [
      {
        "label": "Development",
        "envId": "development",
        "envPrefix": "development",
        "isProduction": false,
        "hosts": [
          {
            "host": "%projectId%.test.mia-platform.eu",
            "scheme": "https"
          },
          {
            "host": "cms.%projectId%.test.mia-platform.eu",
            "isBackoffice": true
          }
        ],
        "cluster": {
          "clusterId": "human-readable-id-of-the-cluster",
          "namespace": "%projectId%-development",
          "kubeContextVariables": {
            "KUBE_URL": "KUBE_DEV_URL",
            "KUBE_TOKEN": "KUBE_DEV_TOKEN"
          }
        }
      }
    ]
  ```

  :::danger
  The values saved here are not encrypted!

  Do not set the `cluster.kubeContextVariables` object with plain values to access the cluster.

>   Instead, prefer the use of an interpolated value. By specifying a variable key name (as shown in the example), the value will be taken from the corresponing project variable. 

Reviewer 2: ```suggestion
  Instead, prefer the use of an interpolated value. By specifying a variable key name (as shown in the example), the value will be taken from the corresponding project variable.
```
